services/construction_service.go

```diff
@@ -1048,10 +1049,37 @@ func encodeMethodArgsStrings(sigData []byte, methodSig string, methodArgs []stri
 				argData = value
 			}
 		case strings.HasPrefix(v, "bytes"):
-			{
-				var value []byte
-				copy(value, methodArgs[i])
+			if v == "bytes" {
+				// Converts dynamically-sized byte array to a slice
+				value, err := hexutil.Decode(methodArgs[i])
+				if err != nil {
+					return nil, err
+				}
 				argData = value
+			} else {
+				// Converts fixed-size byte array (like bytes32) to array
+				sizeStr := strings.TrimPrefix(v, "bytes")
+				size, err := strconv.Atoi(sizeStr)
+				if err != nil {
+					log.Fatal(err)
+				}
+				bytes, err := hexutil.Decode(methodArgs[i])
+				if err != nil {
+					return nil, err
+				}
+				if len(bytes) != size {
+					return nil, fmt.Errorf(
+						"received %d bytes for argument of type %s; expected %d bytes",
+						len(bytes), v, size,
+					)
+				}
+
+				arrayType := reflect.ArrayOf(size, reflect.TypeOf(byte(0)))
+				arrayValue := reflect.New(arrayType).Elem()
+				for i := 0; i < arrayValue.Len() && i < len(bytes); i++ {
```

Comment:
I think we're safe to just put `i < len(bytes)` in the For statement to make it more clean, given `arrayValue` and `bytes` both size are the same, correct ?